### PR TITLE
Ensure repo is always hydrated when needed

### PR DESCRIPTION
**Pull request by asdine**
This PR changes the way the GraphQL RepoResolver accesses the underlying repository by making sure whenever we want to access repository information other than ID and Name, we make sure the inner repo is hydrated.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/git_tree_entry.go [hunk 1]
```diff
@@ -79,9 +79,16 @@ func (r *GitTreeEntryResolver) Content(ctx context.Context) (string, error) {
...
+		repo, err := r.commit.repoResolver.repo(ctx)
 		if err != nil {
 			r.contentErr = err
+			return
+		}
+
+		cachedRepo, err := backend.CachedGitRepo(ctx, repo)
+		if err != nil {
+			r.contentErr = err
+			// Bug?: Should we return here?
```
Comment (asdine) [inline cmd/frontend/graphqlbackend/git_tree_entry.go:91]: This is unrelated to this PR but I realized we are not returning in this particular case, should we?

Comment (unknwon) [reply cmd/frontend/graphqlbackend/git_tree_entry.go:91]: I have no historical context, but I don't see a reason why we shouldn't return.

**Review by unknwon — APPROVED** [review]

File: cmd/frontend/graphqlbackend/repository_git_refs.go [hunk 2]
```diff
@@ -102,7 +102,7 @@ func (r *RepositoryResolver) GitRefs(ctx context.Context, args *refsArgs) (*gitR
 	var tags []*git.Tag
 	if args.Type == nil || *args.Type == gitRefTypeTag {
 		var err error
-		tags, err = git.ListTags(ctx, r.repo.Name)
+		tags, err = git.ListTags(ctx, r.innerRepo.Name)
```
Comment (stefanhengl) [inline cmd/frontend/graphqlbackend/repository_git_refs.go:105]: What about r.repoName() instead?

**Review by stefanhengl — APPROVED** [review]
None of my comments are blockers. I think we should replace instances of `string(r.innerRepo.Name)` with `r.Name`. Feel free to ignore the rest.

File: cmd/frontend/graphqlbackend/repository.go [hunk 3]
```diff
@@ -32,8 +32,10 @@ type RepositoryResolver struct {
 	hydration sync.Once
 	err       error
 
-	repo *types.Repo
-	icon string
+	// innerRepo may only contain ID and Name information.
+	// To access any other repo information, use repo() instead.
+	innerRepo *types.Repo
```
Comment (tsenart) [inline cmd/frontend/graphqlbackend/repository.go:37]: Can you explain the rationale behind calling this innerRepo?

**Review by tsenart — APPROVED** [review]

File: cmd/frontend/graphqlbackend/repository.go (hunk 3 above)
Comment (asdine) [reply cmd/frontend/graphqlbackend/repository.go:37]: `repo` was used by the `repo()` method, so I changed it to that to kind of show that it's not supposed to be used directly. I don't have a strong preference though, I'm open to suggestions

File: cmd/frontend/graphqlbackend/repository_git_refs.go (hunk 2 above)
Comment (asdine) [reply cmd/frontend/graphqlbackend/repository_git_refs.go:105]: Good idea

**Pull request merged**